src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryRegEx.cs

```diff
@@ -0,0 +1,343 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Core.App.ErrorContainers;
+using Microsoft.PowerFx.Core.Binding;
+using Microsoft.PowerFx.Core.Errors;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Syntax;
+using Microsoft.PowerFx.Types;
+using static Microsoft.PowerFx.Core.Localization.TexlStrings;
+
+namespace Microsoft.PowerFx.Functions
+{
+    internal static partial class Library
+    {
+        // https://learn.microsoft.com/en-us/power-platform/power-fx/reference/function-ismatch
+        // In this version of Match, IsMatch and MatchAll functions, predefined patterns are not supported.
+
+        public const string FULLMATCH = "FullMatch";
+        public const string STARTMATCH = "StartMatch";
+        public const string SUBMATCHES = "SubMatches";
+
+        // IsMatch( Text, Pattern [, MatchOptions ] )
+        internal class IsMatchFunction : MatchFunctionBase
```

Comment:
>IsMatchFunction [](http://example.com/codeflow?start=23&length=15)

These functions already exist in Power Apps; can we move it from there instead of redefining it so that we have consistent validation? I understand that this version takes a TimeSpan argument, we can always pass the default value (TimeSpan.Infinite) in Power Apps since it won't be used there.

Reply:
Done and thanks for your help!